robot/impl/local_robot.go

```diff
@@ -171,6 +171,17 @@ func (r *localRobot) Close(ctx context.Context) error {
 	return err
 }
 
+// CloseWait attempts to cleanly close down all constituent parts of the robot, including
+// reconfigureWorkers that may be running outside code and have unexpected behavior. This
+// method should only be used for testing to ensure there are no goroutine leaks.
+func (r *localRobot) CloseWait(ctx context.Context) error {
+	if err := r.Close(ctx); err != nil {
+		return err
+	}
+	r.reconfigureWorkers.Wait()
+	return nil
+}
```

Comment:
exposing this functionality so that we can expose a setup helper to packages outside of `robotimpl`.

Reply:
the robotimpl already has the testing package, if we need to we should just expose the helper

Reply:
per https://github.com/viamrobotics/rdk/pull/4138#discussion_r1657469507 this is no longer necessary.

but just to confirm my understanding, I assumed that we don't use testing functions in any robotimpl functions that are reachable from the main RDK server entrypoint, so therefore tests would be excluded from the server binary - adding a public function that uses a testing package changes that.

that said I haven't done testing around this to confirm and it's unclear how much testing packages add to the finally binary even if it was the case.

Reply:
I don't think that would be the case - the server binary still will never call the testing functions (no code path leads to there), so it wouldn't be packaged (if my understanding of how it works is correct)